core/trino-main/src/main/java/io/trino/execution/SetSessionTask.java

```diff
@@ -110,7 +110,7 @@ public ListenableFuture<Void> execute(
             propertyMetadata.decode(objectValue);
         }
         catch (RuntimeException e) {
-            throw semanticException(INVALID_SESSION_PROPERTY, statement, e.getMessage());
+            throw semanticException(INVALID_SESSION_PROPERTY, statement, "%s", e.getMessage());
```

Comment:
I don't like it. It looks like hacky suppress, I think the previous version is clearer and it should be allowed.

Reply:
But it's not clear to the static analyzers, which want to make sure that the message doesn't malformed format strings, like `%` signs which will error out in `String#format`. It's also not clear to me, because by looking at it I have no idea that it's a format method and I should be wary of possible errors.

There *is* a possibility to add an overload without the format parameters; it won't use `String#format` and it won't be annotated as a format method. But you lose the ability to detect mis-uses of the format method, like:

```java
            throw semanticException(INVALID_SESSION_PROPERTY, statement, "Nested error: " + e.getMessage());
```

which should be:

```java
            throw semanticException(INVALID_SESSION_PROPERTY, statement, "Nested error: %s", e.getMessage());
```

instead.

Reply:
You cannot do this. It is an user facing exception message, so you want the nice and actionable message. This would be a regression.

Reply:
But it doesn't change the message, just makes sure that it won't be misinterpreted as a format string...

Reply:
Maybe `"Invalid session property: %s"`? Or maybe using `semanticException` here is a mistake and we should just `throw new TrinoException`?

Reply:
That's a good point. Changed that to mention the property name as well.